src/config.rs

```diff
@@ -727,6 +731,91 @@ pub fn parse_config_file(buffer: &str, verbosity: u64) -> Args {
             firehosen
         });
 
+        args.kafkas = sinks.get("kafka").map(|snk| {
+            let mut kafkas = Vec::new();
+            for (name, tbl) in snk.as_table().unwrap().iter() {
+                let is_enabled = tbl.get("enabled")
+                    .unwrap_or(&toml::Value::Boolean(true))
+                    .as_bool()
+                    .expect("must be a bool");
+                if !is_enabled {
+                    continue;
+                }
+
+                let mut res = KafkaConfig::default();
+                res.config_path = Some(format!("sinks.kafka.{}", name));
+
+                let topic = tbl.get("topic")
+                    .map(|x| x.as_str().expect("topic must be a string").to_string());
+                if topic.is_none() {
+                    warn!(
+                        "kafka sink {:?} skipped as it does not provide a topic!",
+                        res.config_path
+                    );
+                    continue;
+                }
+                res.topic_name = topic;
+
+                let brokers = tbl.get("brokers").map(|x| {
+                    x.as_str()
+                        .expect("brokers must be a comma-separated list of host or host:port")
+                        .to_string()
+                });
+                if brokers.is_none() {
+                    warn!(
+                        "kafka sink {:?} skipped as it does not provide brokers",
+                        res.config_path
+                    );
+                    continue;
+                }
+                res.brokers = brokers;
+
+                // Allow configuration of librdkafka producer with a sub-table.
+                // Sooo many options:
+                // https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+                res.rdkafka_config = tbl.get("librdkafka").map(|x| {
```

Comment:
Hmm, this is tricky. On the one hand I'm concerned about exposing the underlying implementation through the config: if we choose to use a new library we've suddenly got compatibility issues. On the other hand, there _are_ a ton of configuration options here and it's nice not to have to work them into some abstract cernan state. 

I guess if we _did_ use another library at some point in the future the migration path would be support both, allow config passthrough in a like manner?

Reply:
Yeah, I hear you. On the one hand, we don't necessarily want to co-opt all these settings, nor do we necessarily want the lib users to be stuck with only what we pull out. On the other hand, it's a PITA for those same users if we ever swap out the library. Perhaps if we ever switch libs, we would do what you say wrt to migration path and allow selection of the kafka lib via cargo.toml features?

Reply:
Sounds like a plan to me.